Add tests for Cluster and Network API

dave-tucker opened the pull request:
Network API tests are skipped due to issues mocking the datastore from
ecc. We should look at how to decouple the API from ecc to improve
testability

Signed-off-by: Dave Tucker <dave@socketplane.io>

dave-tucker commented:
@mavenugo let's not merge this until we can find a way to include the network api tests

mavenugo commented:
@dave-tucker Network API tests are enabled via other commits. Can you please update the latest and let me know if we are set for moving this PR along ?

dave-tucker commented:
@mavenugo no I think you were referring to the Network CRUD tests which I enabled a while back... My concern here is for the Network REST API tests that I've added in this commit which I'm currently having to skip. The fixes you pushed *do* allow me to enable at least `TestGetNetworkApi` but the remainder of the tests I've added still fail. This is largely due to the fact that there is no *easy* way to introduce test data in to this datastore. `CreateNetwork` starts creating interfaces, setting addresses, MTUs etc.. which is where my tests are failing right now... as we discussed in Slack, it would be nice to decouple these somehow so things are easier to test

dave-tucker commented:
@mavenugo forgot to add - we can merge this now, but we'll need to open an issue to come back and address the network rest api testing + testability of anything where the data is held in ecc.
